### PR TITLE
Remove more calls to Dart_GetField

### PR DESCRIPTION
**Pull request by abarth**
This patch removes a number of the classes that were generating
Dart_GetField calls, particularly related to drawAtlas and drawVertices.
The functions that used these classes now use primatives instead, which
is more efficient.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @chinmaygarde

**Comment by chinmaygarde** [comment]
lgtm

File: sky/engine/core/dart/painting.dart [hunk 1]
```diff
@@ -703,59 +719,108 @@ class Canvas extends NativeFieldWrapperClass2 {
...
-                     List<int> indicies,
+                     Float32List vertices,
+                     Float32List textureCoordinates,
+                     Int32List colors,
+                     int transferMode,
+                     Int32List indicies,
                      Paint paint) native "Canvas_drawVertices";
 
-  void drawAtlas(Image image,
+  // TODO(eseidel): Paint should be optional, but optional doesn't work.
```
Comment (Hixie) [inline sky/engine/core/dart/painting.dart]: you could make the paint optional on the dart side now

File: sky/engine/core/dart/painting.dart [hunk 2]
```diff
@@ -703,59 +719,108 @@ class Canvas extends NativeFieldWrapperClass2 {
...
                  List<Rect> rects,
                  List<Color> colors,
-                 TransferMode mode,
+                 TransferMode transferMode,
                  Rect cullRect,
                  Paint paint) {
-    if (transforms.length != rects.length)
+    final int rectCount = rects.length;
+
+    if (transforms.length != rectCount)
```
Comment (Hixie) [inline sky/engine/core/dart/painting.dart]: same discussions regarding nulls here

**Comment by Hixie** [comment]
LGTM

**Pull request merged**